pkg/install/genevalogging.go

```diff
@@ -0,0 +1,399 @@
+	        </Accounts>
+	        <Management eventVolume="Large" defaultRetentionInDays="90">
+	            <Identity tenantNameAlias="ResourceName" roleNameAlias="ResourceGroupName" roleInstanceNameAlias="SubscriptionId">
+	                <IdentityComponent name="Region">{{ .Region | Escape }}</IdentityComponent>
+	                <IdentityComponent name="SubscriptionId">{{ .SubscriptionId | Escape }}</IdentityComponent>
+	                <IdentityComponent name="ResourceGroupName">{{ .ResourceGroupName | Escape }}</IdentityComponent>
+	                <IdentityComponent name="ResourceName">{{ .ResourceName | Escape }}</IdentityComponent>
+	            </Identity>
+	            <AgentResourceUsage diskQuotaInMB="50000"/>
+	        </Management>
+	        <Sources>
+	            <Source name="journald" dynamic_schema="true"/>
+	            <Source name="audit" dynamic_schema="true"/>
+	        </Sources>
+	        <Events>
+	            <MdsdEvents>
+	                <MdsdEventSource source="journald">
+	                    <RouteEvent eventName="CustomerSyslogEvents" storeType="CentralBond" priority="High"/>
+	                </MdsdEventSource>
+	                <MdsdEventSource source="audit">
+	                    <RouteEvent eventName="CustomerAuditLogEvents" storeType="CentralBond" priority="High"/>
+	                </MdsdEventSource>
+	            </MdsdEvents>
+	        </Events>
+		</MonitoringManagement>`
+
+	t := template.Must(template.New("").Funcs(map[string]interface{}{
+		"Escape": func(s string) (string, error) {
+			var b bytes.Buffer
+			err := xml.EscapeText(&b, []byte(s))
+			return b.String(), err
+		},
+	}).Parse(tmpl))
+
+	b := &bytes.Buffer{}
+
+	err := t.Execute(b, map[string]string{
+		"Namespace":         genevaLoggingNamespace,
+		"Account":           genevaLoggingAccount,
+		"Region":            location,
+		"SubscriptionId":    subscriptionID,
+		"ResourceName":      clusterName,
+		"ResourceGroupName": resourceGroup,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return string(b.Bytes()), nil
+}
+
+func (i *Installer) installGenevaLogging(ctx context.Context) error {
+	g, err := i.loadGraph(ctx)
+	if err != nil {
+		return err
+	}
+
+	installConfig := g[reflect.TypeOf(&installconfig.InstallConfig{})].(*installconfig.InstallConfig)
+	resourceGroup := i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID[strings.LastIndexByte(i.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')+1:]
+	mdsdConf, err := mdsdConfig(installConfig.Config.Azure.Region, resourceGroup, "subID", installConfig.Config.ClusterName, "GenevaLoggingNamespace", "GenevaLoggingAccount")
+	if err != nil {
+		return err
+	}
+
+	restConfig, err := restconfig.RestConfig(ctx, i.env, i.doc.OpenShiftCluster)
+	if err != nil {
+		return err
+	}
+
+	cli, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+	_, err = cli.CoreV1().Namespaces().Create(&v1.Namespace{
```

Comment:
there's bound to be a race here - after you create a namespace, OpenShift controllers asynchronously put stuff in there (service accounts, rbac rules, namespace annotations)...  Some of which (not quite sure which) might be necessary for the DaemonSet to create and function properly.  Likely it'll be necessary to wait for something.